common/web/gesture-recognizer/src/engine/headless/trackedPoint.ts

```diff
@@ -25,68 +27,62 @@ export class TrackedPoint {
+  private _path: TrackedPath<HoveredItemType>;
 
-  private _path: TrackedPath;
+  private static _jsonIdSeed: -1;
 
   /**
    * Tracks the coordinates and timestamps of each update for the lifetime of this `TrackedPoint`.
    */
-  public get path(): TrackedPath {
+  public get path(): TrackedPath<HoveredItemType> {
     return this._path;
   }
 
   /**
    * Constructs a new TrackedPoint instance for tracking updates to an active input point over time.
    * @param identifier     The system identifier for the input point's events.
-   * @param initialTarget  The initiating event's original target element
+   * @param initialHoveredItem  The initiating event's original target element
    * @param isFromTouch    `true` if sourced from a `TouchEvent`; `false` otherwise.
    */
-  constructor(identifier: number,
-              initialTarget: EventTarget,
-              isFromTouch: boolean);
+  constructor(identifier: number, isFromTouch: boolean) {
+    this.rawIdentifier = identifier;
+    this.isFromTouch = isFromTouch;
+    this._path = new TrackedPath();
+  }
+
   /**
    * Deserializes a TrackedPoint instance from its serialized-JSON form.
+   * @param jsonObj  The JSON representation to deserialize.
    * @param identifier The unique identifier to assign to this instance.
-   * @param parsedObj  The JSON representation to deserialize.
    */
-  constructor(identifier: number,
-    parsedObj: JSONTrackedPoint);
-  constructor(identifier: number,
-              obj: EventTarget | JSONTrackedPoint,
-              isFromTouch?: boolean) {
-    this.rawIdentifier = identifier;
-    if(obj instanceof EventTarget) {
-      this._initialTarget = obj;
-      this.isFromTouch = isFromTouch;
-      this._path = new TrackedPath();
-    } else {
-      // // TEMP:  conversion of old format.
-      // if(obj['sequence']) {
-      //   obj = obj['sequence'];
-      // }
-      // @ts-ignore
-      this.isFromTouch = obj.isFromTouch;
+  public static deserialize(jsonObj: JSONTrackedPoint, identifier: number) {
+    const id = identifier !== undefined ? identifier : this._jsonIdSeed++;
+    const isFromTouch = jsonObj.isFromTouch;
+    const path = TrackedPath.deserialize(jsonObj.path);
+
+    const instance = new TrackedPoint(id, isFromTouch);
+    instance._path = path;
+    return instance;
+  }
 
-      // TEMP:  conversion of old format
-      // @ts-ignore
-      let path = obj.path;
-      // if(obj['samples']) {
-      //   // @ts-ignore
-      //   path = {
-      //     coords: obj['samples']
-      //   };
-      // }
+  public update(sample: InputSample<HoveredItemType>) {
+    this.path.extend(sample);
+  }
 
-      this._path = new TrackedPath(path);
-    }
+  /**
+   * The identifying metadata returned by the configuration's specified `itemIdentifier` for
+   * the target of the first `Event` that corresponded to this `TrackedPoint`.
+   */
+  public get initialHoveredItem(): HoveredItemType {
+    return this.path.coords[0].item;
```

Comment:
This is going to throw if ever called when the object has an empty path. Is that the failure path we want here, or should it be returning null in that case for the caller to deal with?